cEpiabm/src/reporters/percell_compartment_reporter.cpp

```diff
@@ -0,0 +1,88 @@
+
+#include "percell_compartment_reporter.hpp"
+#include "../logfile.hpp"
+
+
+namespace epiabm
+{
+
+
+    PerCellCompartmentReporter::PerCellCompartmentReporter(const std::string folder) :
+        TimestepReporterInterface(folder, true),
+        m_compartments(
+            {
+            InfectionStatus::Susceptible,
+            InfectionStatus::Exposed,
+            InfectionStatus::InfectASympt,
+            InfectionStatus::InfectMild,
+            InfectionStatus::InfectGP,
+            InfectionStatus::InfectHosp,
+            InfectionStatus::InfectICU,
+            InfectionStatus::InfectICURecov,
+            InfectionStatus::Dead,
+            InfectionStatus::Recovered
```

Comment:
I don't like that we have to define these in the `cell_compartment_reporter` and the `percell_compartment_reporter` - is there any way to inherit/import these from a single place (so that we would only have to change them in one place if we wanted to change the list of compartments

Reply:
Could create a global variable which stores a list of all statuses.

Reply:
Yeah that definitely seems tidier and less error-prone/easier for the future if that's easy to do!